File: backend/app/schemas/sales.py
from pydantic import BaseModel, Field
from typing import List
from decimal import Decimal

class CreateSaleItem(BaseModel):
    product_id: int = Field(
        json_schema_extra={"example": 1}
    )
    quantity: int = Field(
        gt=0,
        json_schema_extra={"example": 2}
    )
    unit_price: Decimal = Field(
        ge=0,
        json_schema_extra={"example": "100.00"}
    )


class CreateSaleRequest(BaseModel):
    sales_channel: str = Field(
        default="OFFLINE",
        json_schema_extra={"example": "ONLINE"}
    )
    items: List[CreateSaleItem]


class CreateSaleResponse(BaseModel):
    order_id: int
    total_amount: Decimal

File: frontend/src/types/sales.ts
export interface CreateSaleItemRequest {
  product_id: number
  quantity: number
  unit_price: number
}

export interface CreateSaleRequest {
  sales_channel: string
  items: CreateSaleItemRequest[]
}

export interface CreateSaleResponse {
  order_id: number
  total_amount: string
}

export interface SaleRead {
  id: number
  order_date: string
  sales_channel: string
  total_amount: string
}

export interface SaleItemRead {
  product_id: number
  quantity: number
  unit_price: string
  line_total: string
}

export interface SaleDetailRead {
  id: number
  order_date: string
  sales_channel: string
  total_amount: string
  items: SaleItemRead[]
}

export interface ListSalesParams {
  date_from?: string
  date_to?: string
  sales_channel?: string
  product_id?: number
}
